frontend/dialogs/OBSBasicSourceSelect.cpp

```diff
@@ -15,100 +16,70 @@
 };
 
-bool OBSBasicSourceSelect::EnumSources(void *data, obs_source_t *source)
-{
-	if (obs_source_is_hidden(source))
-		return true;
-
-	OBSBasicSourceSelect *window = static_cast<OBSBasicSourceSelect *>(data);
-	const char *name = obs_source_get_name(source);
-	const char *id = obs_source_get_unversioned_id(source);
-
-	if (strcmp(id, window->id) == 0)
-		window->ui->sourceList->addItem(QT_UTF8(name));
-
-	return true;
-}
-
-bool OBSBasicSourceSelect::EnumGroups(void *data, obs_source_t *source)
+namespace {
+QString getSourceDisplayName(QString type)
 {
-	OBSBasicSourceSelect *window = static_cast<OBSBasicSourceSelect *>(data);
-	const char *name = obs_source_get_name(source);
-	const char *id = obs_source_get_unversioned_id(source);
-
-	if (strcmp(id, window->id) == 0) {
-		OBSBasic *main = OBSBasic::Get();
-		OBSScene scene = main->GetCurrentScene();
-
-		obs_sceneitem_t *existing = obs_scene_get_group(scene, name);
-		if (!existing)
-			window->ui->sourceList->addItem(QT_UTF8(name));
+	if (type == "scene") {
+		return QTStr("Basic.Scene");
 	}
 
-	return true;
-}
+	const char *inputChar = obs_get_latest_input_type_id(type.toUtf8().constData());
+	const char *displayChar = obs_source_get_display_name(inputChar);
+	std::string displayId = (displayChar) ? displayChar : "";
 
-void OBSBasicSourceSelect::OBSSourceAdded(void *data, calldata_t *calldata)
-{
-	OBSBasicSourceSelect *window = static_cast<OBSBasicSourceSelect *>(data);
-	obs_source_t *source = (obs_source_t *)calldata_ptr(calldata, "source");
-
-	QMetaObject::invokeMethod(window, "SourceAdded", Q_ARG(OBSSource, source));
-}
-
-void OBSBasicSourceSelect::OBSSourceRemoved(void *data, calldata_t *calldata)
-{
-	OBSBasicSourceSelect *window = static_cast<OBSBasicSourceSelect *>(data);
-	obs_source_t *source = (obs_source_t *)calldata_ptr(calldata, "source");
-
-	QMetaObject::invokeMethod(window, "SourceRemoved", Q_ARG(OBSSource, source));
-}
-
-void OBSBasicSourceSelect::SourceAdded(OBSSource source)
-{
-	const char *name = obs_source_get_name(source);
-	const char *sourceId = obs_source_get_unversioned_id(source);
-
-	if (strcmp(sourceId, id) != 0)
-		return;
-
-	ui->sourceList->addItem(name);
+	if (!displayId.empty()) {
+		return QString::fromStdString(displayId);
+	} else {
+		return QString();
+	}
 }
 
-void OBSBasicSourceSelect::SourceRemoved(OBSSource source)
+std::string getNewSourceName(std::string_view name)
 {
-	const char *name = obs_source_get_name(source);
-	const char *sourceId = obs_source_get_unversioned_id(source);
+	std::string newName(name);
 
-	if (strcmp(sourceId, id) != 0)
-		return;
+	int suffix = 1;
 
-	QList<QListWidgetItem *> items = ui->sourceList->findItems(name, Qt::MatchFixedString);
+	for (;;) {
+		OBSSourceAutoRelease existing_source = obs_get_source_by_name(newName.c_str());
+		if (!existing_source) {
+			break;
+		}
 
-	if (!items.count())
-		return;
+		newName = name;
+		newName += " ";
+		newName += std::to_string(suffix);
+	}
```

Comment:
The suffix is never incremented as far as I can tell, so this would result in the following behaviour:

* Requested name is "MySource"
* Source with that name already exists
* Name is changed to "MySource 1"
* If "MySource 1" also already exists, the name is changed to "MySource 1 1"

I'm of the opinion that we should not do this in the first place: If a name is taken, just require users to choose another name. Automatic suffixing makes sense for bulk operations where it would be more frustrating for a user to provide alternatives for every collision (e.g. when copying a lot of files into a directory with existing files) but for simple operations like these it makes more sense to immediately prompt for an alternative name.